pkg/pillar/Dockerfile

```
@@ -179,6 +179,8 @@ SHELL ["/bin/sh", "-c"]
 
 COPY --from=collector /out /
 
+RUN rm -rf /usr/bin/qemu-storage-daemon
```

Does it make sense to already remove it in the `collector` container image?

---

```suggestion
RUN rm -f /usr/bin/qemu-storage-daemon
```
this way the build fails in case the qemu package changes and f.e. they change the path of it

---

Maybe. In the very beginning of this work, I tried to avoid unnecessary builds/downloads at the very moment they are introduced in a Dockerfile. But im most cases it was not that straightforward, basically because of very unexpected dependencies. So, I gave up and decided to remove the unnecessary files on the very last stage of building. That's true for all the changes in different Dockerfiles across this PR.

---

It is even more true for the firmware packages; first they are copied and then they're removed again if the architecture does not match.

> and decided to remove the unnecessary files on the very last stage of building

And I fear this will make it even less straight-forward for the next person that touches it.

Why did you decide against just increasing the image size... ;-)

---

Ibelieve you are right, but I don't know how to do the removal with a proper accuracy. It would like someone with better experience in firmware integration to help here.
Until the then, I think it's still a good enough solution to free some space.
I would like to avoid just increasing the partition size all the time. It will be very soon when this approach reaches the limit. So, we have to start doing something else: either cleaning stuff, or by introducing some after-Installation steps to deliver some parts of the system not from the installer partition. For now, let's just do the first...

---

But you know that here is already:
> work on a generic solution to reduce eve image size

(https://github.com/lf-edge/eve/pull/4553#issuecomment-2635524087 )

---

(feel free to resolve)